Purchases/Caching/DeviceCache.swift

```diff
@@ -493,4 +499,38 @@ fileprivate extension UserDefaults {
 
 }
 
+private extension DeviceCache {
+    enum CacheDuration {
+        // swiftlint:disable:next nesting
+        enum AppStatus {
+            case foreground
+            case background
+
+            init(backgrounded: Bool) {
+                self = backgrounded ? .background : .foreground
+            }
+        }
+
+        // swiftlint:disable:next nesting
+        enum Environment {
+            case production
+            case sandbox
+
+            init(sandbox: Bool) {
+                self = sandbox ? .sandbox : .production
+            }
+        }
+
+        static func duration(status: AppStatus, environment: Environment) -> TimeInterval {
+            switch (environment, status) {
+            case (.production, .foreground): return 60 * 5.0
+            case (.production, .background): return 60 * 60 * 25.0
+
+            case (.sandbox, .foreground): return 60 * 5.0
+            case (.sandbox, .background): return 60 * 5.0
```

Comment:
Also I decided to duplicate this instead of using one `case` because we might choose different durations.